src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,59 @@
+package duke;
+
+import java.io.IOException;
+
+import duke.command.Command;
+import duke.task.TaskList;
+
+public class Duke {
+    private Storage storage;
+    private TaskList taskList;
+    private Ui ui;
+
+    /**
+     * Creates a Duke instance with filePath to where data is stored.
+     *
+     * @param filePath
+     */
+    public Duke(String filePath) throws IOException {
+        storage = new Storage(filePath);
+        ui = new Ui();
+        try {
+            taskList = storage.load();
+        } catch (DukeException e) {
```

Comment:
According to coding standard, it's not good to have single letter variables. Maybe this can be called "exception" instead?